.editorconfig

```diff
@@ -0,0 +1,718 @@
+root = true
```

Comment:
How is this file used?  Is it actually needed?

Reply:
I saw @stIncMale added this file into `mongo-java-driver`. It seems an easy way to ensure code format issue for .editorconfig is supported universally, including Intellij IDE (see https://www.jetbrains.com/help/idea/editorconfig.html).

I think using it could ensure all developers are on the same pages when it comes to code styles issue. It is a pity Hibernate ORM project didn't use it, but vast majority of other open-source projects use it.